report/src/components/Modals/DetailsButtonModal.js

```diff
@@ -0,0 +1,246 @@
+export const DetailsButtonModal = ({ lines, summary }) => {
+  const implementationArray = lines.filter((element) => element.implementation);
+  const caseArray = lines.filter((element) => element.case);
+
+  function TestStatus(implementationImage) {
+    let testStatusArray;
+    var arrayOfSeqCases = [];
+    implementationArray.forEach((seqImplementation) => {
+      if (seqImplementation.implementation === implementationImage) {
+        // var caseResults = [];
+        if (seqImplementation.skipped) {
+          var seq = seqImplementation.seq;
+          caseArray.forEach((seqCase) => {
+            if (seqCase.seq == seq) {
+              var hasResultsArray = false;
+              arrayOfSeqCases.push(seqCase);
+              testStatusArray = ["skipped", seqImplementation, hasResultsArray];
+            }
+          });
+        } else if (seqImplementation.caught) {
+          let seq = seqImplementation.seq;
+          caseArray.forEach((seqCase) => {
+            if (seqCase.seq == seq) {
+              var hasResultsArray = true;
+              arrayOfSeqCases.push(seqCase);
+              testStatusArray = [
+                "unexpectedlyErrored",
+                seqImplementation,
+                hasResultsArray,
+              ];
+            }
+          });
+        } else if (seqImplementation.results && seqImplementation.expected) {
+          var caseResults = seqImplementation.results.filter(
+            (element) => element.skipped,
+          );
+          if (caseResults.length > 0) {
+            var seq = seqImplementation.seq;
+            caseArray.forEach((seqCase) => {
+              if (seqCase.seq == seq) {
+                var hasResultsArray = true;
+                arrayOfSeqCases.push(seqCase);
+                testStatusArray = [
+                  "skipped",
+                  seqImplementation,
+                  hasResultsArray,
+                ];
+              }
+            });
+          }
+
+          if (
+            seqImplementation.results.every(
+              (element) =>
+                typeof element === "object" &&
+                Object.keys(element).length === 1 &&
+                "valid" in element,
+            )
+          ) {
+          }
+        } else if (seqImplementation.results) {
+          var caseResults = seqImplementation.results.filter(
+            (seqImplementation) => seqImplementation.context,
+          );
+          // console.log(caseResults);
```

Comment:
More dead code sprinkled in this file -- good to do a pass and just look for any lines you've commented out -- perhaps also look at whether eslint can raise warnings for such things, I suspect/hope it can, so we should enable that -- let me know if you want help figuring out how.

Reply:
Cleaned it up!

Reply:
> perhaps also look at whether eslint can raise warnings for such things...

Ya please help me to understand how ESLint works.

Reply:
Definitely! We can probably hit this too in a follow-up PR, no need to block this one, but yeah I'll show you how it works!